File: gtServer/Objects/GuitarString.cs
using gtServer.Enums;

namespace gtServer.Objects;

public class GuitarString
{
    public List<ChromaticScale> CSNotes { get; set; }
    public List<string> SNotes { get; set; }
    public int NumFrets { get; set; }
    public GuitarString(int numFrets = 24)
    {
        NumFrets = numFrets;
        CSNotes = [];
        SNotes = [];
    }
    public void TuneString(ChromaticScale note)
    {
        List<ChromaticScale> csNotes = [];
        List<string> sNotes = [];
        ChromaticScale nextNote = note;

        for (int i = 0; i < NumFrets; i++)
        {
            csNotes.Add(nextNote);
            sNotes.Add(ChromaticScaleConversions.NoteStringValue(nextNote));
            nextNote = ScalePatterns.FindNextNote(nextNote);
        }
        CSNotes = csNotes;
        SNotes = sNotes;
    }
}

File: fastapi-server/objects/guitar_string.py
from objects.scale_patterns import find_next_chromatic


class GuitarString():
    def __init__(self, num_frets: int = 24):
        self.num_frets: int = num_frets
        self.frets: list[str] = []

    def tune_string(self, note: str):
        self.frets = []
        for _ in range(0, self.num_frets):
            self.frets.append(note)
            note = find_next_chromatic(note)
